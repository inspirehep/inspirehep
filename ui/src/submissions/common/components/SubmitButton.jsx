import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { Button } from 'antd';
import usePrevious from '../../../common/hooks/usePrevious';

function SubmitButton({ isSubmitting, isValidating, isValid }) {
  const previousIsSubmitting = usePrevious(isSubmitting);

  useEffect(
    () => {
      const hasTriedToSubmitInvalidForm =
        previousIsSubmitting && !isSubmitting && !isValid;
      if (hasTriedToSubmitInvalidForm) {
        window.scrollTo(0, 0);
      }
    },
    [isSubmitting, isValid, previousIsSubmitting]
  );

  return (
    <Button
      type="primary"
      htmlType="submit"
      loading={isSubmitting || isValidating}
    >
      Submit
    </Button>
  );
}

SubmitButton.propTypes = {
  isSubmitting: PropTypes.bool.isRequired,
  isValidating: PropTypes.bool.isRequired,
  isValid: PropTypes.bool.isRequired,
};

export default SubmitButton;
