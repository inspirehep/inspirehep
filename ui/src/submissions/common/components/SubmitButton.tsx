import React, { useEffect } from 'react';
import { Button } from 'antd';
import { useFormikContext } from 'formik';

import usePrevious from '../../../common/hooks/usePrevious';

function SubmitButton() {
  const { isSubmitting, isValid, isValidating } = useFormikContext();
  const previousIsSubmitting = usePrevious(isSubmitting);
  useEffect(
    () => {
      const hasTriedToSubmitInvalidForm =
        previousIsSubmitting && !isSubmitting && !isValid;
      if (hasTriedToSubmitInvalidForm) {
        window.scrollTo(0, 0);
      }
    },
    [isSubmitting, isValid, previousIsSubmitting]
  );

  return (
    <Button
      type="primary"
      htmlType="submit"
      loading={isSubmitting || isValidating}
    >
      Submit
    </Button>
  );
}

export default SubmitButton;
