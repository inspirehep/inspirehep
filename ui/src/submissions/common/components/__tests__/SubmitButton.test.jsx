import React from 'react';
import { shallow } from 'enzyme';
import { useFormikContext } from 'formik';
import SubmitButton from '../SubmitButton';

jest.mock('formik');

describe('SubmitButton', () => {
  it('renders with all props set', () => {
    const contextValue = {
      isValid: true,
      isSubmitting: false,
      isValidating: false,
    };
    useFormikContext.mockImplementation(() => contextValue);
    const wrapper = shallow(<SubmitButton />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with loading', () => {
    const contextValue = {
      isValid: true,
      isSubmitting: true,
      isValidating: false,
    };
    useFormikContext.mockImplementation(() => contextValue);
    const wrapper = shallow(<SubmitButton />);
    expect(wrapper).toMatchSnapshot();
  });
});
