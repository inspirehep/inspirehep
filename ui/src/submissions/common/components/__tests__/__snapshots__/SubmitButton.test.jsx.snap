// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubmitButton renders with all props set 1`] = `
<Button
  data-test-id="submit-button"
  htmlType="submit"
  loading={false}
  type="primary"
>
  Submit
</Button>
`;

exports[`SubmitButton renders with loading 1`] = `
<Button
  data-test-id="submit-button"
  htmlType="submit"
  loading={true}
  type="primary"
>
  Submit
</Button>
`;
